training/src/app/modules/chuck-norris/components/joke-generator/joke-generator.component.scss

```diff
@@ -0,0 +1,92 @@
+@import "../../../../../styles";
+
+.search-style {
+  display: grid;
+  grid-template-columns:  1fr 0.3fr;
+  grid-gap: 70px;
+  background-color: $off-white-color;
+  border-radius: 0.4em;
+  margin: 8%;
+  width: 80%;
+  padding: 3% 3% 1% 3%;
+
+  .key-word-style {
+    display: grid;
+    grid-template-columns:  0.21fr 0.7fr;
+  }
+}
+
+.random-container {
+  display: grid;
+  grid-template-columns:  1fr 1fr 0.7fr;
+  grid-gap: 10px;
+  background-color: $off-white-color;
+  border-radius: 0.4em;
+  margin: 8%;
+  width: 80%;
+  padding: 3% 3% 1% 3%;
+
+  mat-select {
+    background-color: $white-color;
+    height: 40px;
+    border-radius: 5px;
+  }
+
+  .your-name-style {
+    display: grid;
+    grid-gap: 10px;
+    grid-template-columns:  0.25fr 0.9fr;
+  }
+
+  .category-style {
+    display: grid;
+    grid-template-columns:  0.1fr 0.9fr;
+    grid-gap: 10px;
+  }
+}
+
+a {
+  padding-top: 12%;
```

Comment:
we avoid giving padding/margins/width perctenage , unless its for example 50% or 100% of the screen.
if u want to center it this is not a good way